### PR TITLE
EI-179/EI-728 Add GitHub actions to build and publish NPM package

### PR DESCRIPTION
**Pull request by Evolter**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the tag name
+        run: echo "TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+      - run: echo "'${{ env.TAG }}'"
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14.18.1
```
Comment (mihai-dinculescu) [inline .github/workflows/publish.yml:17]: Do we need a specific node version here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+### Added
+- Initial release of `iotics-grpc-client-ts`.
+
+[unreleased]: https://github.com/Iotic-Labs/iotics-grpc-client-ts/compare/v0.0.0...HEAD
```
Comment (mihai-dinculescu) [inline CHANGELOG.md:11]: This should be a link to the tag, not to the diff.

File: Makefile [hunk 3]
```diff
@@ -39,15 +39,14 @@ list:
 	$(BUF) ls-files
 
 publish:
-	@echo "WIP"
+	yarn publish --access public
```
Comment (mihai-dinculescu) [inline Makefile:42]: Do we need this behind make?

**Review by mihai-dinculescu — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (Evolter) [reply .github/workflows/publish.yml:17]: Most of the example I've seen used just a major version here (i.e. 14)
I just used specific version to match portal, but I don't think it needs to be like that.
We could also use version 16, although I tested only 14 so far.

File: Makefile (hunk 3 above)
Comment (Evolter) [reply Makefile:42]: Agree, I often use bespoke commands myself, but having an endpoint anyone can use and expect the same result as others is something nice to have IMHO, so if no objections I'm leaving it as is.

**Pull request merged**